### PR TITLE
OculusDevice improvements

### PR DESCRIPTION
**Pull request by rickyviking**
1. exposed full ovrHmdDescription: this is needed for instance to distinguish which Oculus model is currently in use
2. added methods to modify near/far planes after creating the Oculus device: this allow an application to modify the value at runtime, depending on the loaded scene. Should be safe as the values are used on a frame basis to compute the projection matrix.

### CODE REVIEW

**Comment by bjornblissing** [comment]
1. Even though I can see the benefit of this, I do have some (minor) objections to exposing the full `ovrHmdDescription` struct. My goal is to hide as much of the internal Oculus implementation as possible. As of now the only exposed OculusSDK structs are in the code below. These could probably be wrapped as well in the future. So I would rather have functions to access the relevant fields in the  `ovrHmdDescription` struct and make them public as builtin C++ data types.
https://github.com/bjornblissing/osgoculusviewer/blob/4c8a3d92f25e3698bfbcf76c10e3775f875ed321/include/oculusdevice.h#L81-L93

2. This on the other hand seems sensible.

File: include/oculusdevice.h [hunk 1]
```diff
@@ -44,13 +44,21 @@ class OculusDevice : public osg::Referenced {
...
   float nearClip() const {
     return m_nearClip;
   }
   float farClip() const {
     return m_farClip;
   }
 
+  // Added these methods to modify near and far clip after constructing the device.
+  // As these are used on a frame basis only to compute the projection matrix through 
+  // the Oculus SDK, it's safe to modify them any time at runtime.
```
Comment (bjornblissing) [inline include/oculusdevice.h:58]: Remove excessive comments.

**Comment by rickyviking** [comment]
I see. The fact is that also the Type field of the description I'm reading is in turn an enum belonging to the Oculus SDK. 
Wrapping that enum (which clearly belongs to the SDK) seems a bit of a stretch to me.
Let me know what you think, in case I can update the PR with the second point only.

**Comment by bjornblissing** [comment]
I split the changes into:
1f8ec5ed30dbde9e2aded4ddf6018593dcd67dfb
962690e09f5ea5df61c0ffd310090633d0bf5c44

**Pull request closed without merging**